packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,739 @@
+				let tree: ISharedTree;
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree as JS Object (${TestPrimitives[dataType]}): manipulations with ${i} nodes`,
+					before: async () => {
+						tree = getTestTreeAsJSObject(i, TreeShape.Wide, dataType);
+					},
+					benchmarkFn: () => {
+						manipulateTreeAsJSObject(tree, dataType, random);
+					},
+				});
+			}
+		}
+	});
+	describe("Cursors", () => {
+		for (let dataType = 0 as TestPrimitives; dataType <= 4; dataType++) {
+			for (const i of deepTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Deep Tree (${TestPrimitives[dataType]}) with cursor: reads with ${i} nodes`,
+					before: async () => {
+						tree = await generateTestTree(i, TreeShape.Deep, dataType, random);
+					},
+					benchmarkFn: () => {
+						readCursorTree(tree.forest, i, TreeShape.Deep);
+					},
+				});
+			}
+			for (const i of wideTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree (${TestPrimitives[dataType]}) with cursor: reads with ${i} nodes`,
+					before: async () => {
+						tree = await generateTestTree(i, TreeShape.Wide, dataType, random);
+					},
+					benchmarkFn: () => {
+						readCursorTree(tree.forest, i, TreeShape.Wide);
+					},
+				});
+			}
+			for (const i of deepTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Deep Tree (${TestPrimitives[dataType]}) with cursor: writes ${i} nodes`,
+					before: () => {},
+					benchmarkFn: async () => {
+						tree = await generateTestTree(i, TreeShape.Deep, dataType, random);
+					},
+				});
+			}
+			for (const i of wideTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree (${TestPrimitives[dataType]}) with cursor: writes ${i} nodes`,
+					before: () => {},
+					benchmarkFn: async () => {
+						tree = await generateTestTree(i, TreeShape.Wide, dataType, random);
+					},
+				});
+			}
+			for (const i of deepTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Deep Tree (${TestPrimitives[dataType]}) with cursor: manipulations with ${i} nodes`,
+					before: async () => {
+						tree = await generateTestTree(i, TreeShape.Deep, dataType, random);
+					},
+					benchmarkFn: () => {
+						manipulateTree(tree, i, TreeShape.Deep, dataType, random);
+					},
+				});
+			}
+			for (const i of wideTestSizes) {
+				let tree: ISharedTree;
+				const random = makeRandom(0);
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Wide Tree (${TestPrimitives[dataType]}) with cursor: manipulations with ${i} nodes`,
```

Comment:
Ideally someone can understand what a benchamrk means when they look at its performance number and title.

Suppose this benchmark tool 1 second. That tells me we can do what in a second? Do a manipulation with x nodes with a cursor in a wide tree? Thats close to enough information, but what is the specific manuliplation this is tersting, and it doesn't mention the size of the tree. Also, cursors arn't an API for manuliplating treres,: there arer just a reading API, so thats a bit confusing too.

A test named something like "Update value in X node wide tree" or "Update value at leaf of X node deep tree" (and have those just update a single value per iteration) would be real clear. Then we could put those in a suite names something like "Edit using Editable Tree" or "Edit using Transaction" or "Edit forest using deltas" or "Edit JS object"

Reply:
Updated benchmark titles to better describe what they are doing